### PR TITLE
feat: Hookup devcard page to real data

### PR DESCRIPTION
**Pull request by foxyblocks**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

- Adds live data for dev cards
- Migrates the card image generation endpoint into the insights repo 
- Implements "logged-in" view [from the figma](https://www.figma.com/file/hF1wdcQDIcgWtpbqRIEzNB/DevCards?type=design&node-id=585%3A2165&mode=dev), (minus the github button which has been de-scoped)
- 
## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/insights/assets/555044/2c6cb0e6-e7bf-4876-adb8-34bb41b3d68b

https://github.com/open-sauced/insights/assets/555044/3b51592a-dffd-41cc-bab1-25f44069235c

CleanShot 2023-06-28 at 13 35 02@2x
CleanShot 2023-06-28 at 13 36 38@2x

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [x] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: pages/user/[username]/card.tsx [hunk 1]
```diff
@@ -1,48 +1,167 @@
 import { GetServerSideProps, NextPage } from "next";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import { useTransition, animated } from "react-spring";
 import Button from "components/atoms/Button/button";
 import HeaderLogo from "components/molecules/HeaderLogo/header-logo";
-import DevCardCarousel from "components/organisms/DevCardCarousel/dev-card-carousel";
-import { STUB_DEV_CARDS } from "components/organisms/DevCardCarousel/stubData";
+import DevCardCarousel, { DevCardCarouselProps } from "components/organisms/DevCardCarousel/dev-card-carousel";
```
Comment (brandonroberts) [inline pages/user/[username]/card.tsx:6]: nit: Will you clean up the imports where third-party imports (react, next, cntl, etc) are at the top, followed by local imports (not a blocker)

Comment (foxyblocks) [reply pages/user/[username]/card.tsx:6]: Yah, i think i can turn this into a eslint rule too

File: lib/utils/urls.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  url.searchParams.append("text", "Check out my Open Sauced DevCard!");
+  url.searchParams.append("via", "saucedopen");
+  url.searchParams.append("url", cardPageUrl(username));
+  return url.toString();
+};
+
+export const linkedinCardShareUrl = (username: string, summary: string) => {
+  const url = new URL("https://www.linkedin.com/shareArticle?mini=true");
+  url.searchParams.append("url", cardPageUrl(username));
+  url.searchParams.append("title", "Check out my Open Sauced DevCard!");
```
Comment (bdougie) [inline lib/utils/urls.ts:37]:
```suggestion
  url.searchParams.append("title", "Check out my open-source contributions card!");
```

Suggesting different copy. Not sure we want to call the "Devcards" publicly. We can update this later after some testing.

File: lib/utils/urls.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ *  DevCard URLs
+ * -------------------------------------------------------------------------------
+ */
+export const cardPageUrl = (username: string) => siteUrl(`user/${username}/card`);
+
+export const cardImageUrl = (username: string) => siteUrl(`api/card.png?username=${username}`);
+
+export const twitterCardShareUrl = (username: string) => {
+  const url = new URL("https://twitter.com/intent/tweet");
+  url.searchParams.append("text", "Check out my Open Sauced DevCard!");
```
Comment (bdougie) [inline lib/utils/urls.ts:28]:
```suggestion
  url.searchParams.append("text", "Check out my open-source contributions card!");
```

**Review by bdougie — APPROVED** [review]
QA and functionality, works for me!

Updating the onboarding to include cards is next up, then the 404-page experience. 

Related to this performance stuff, would leveraging edge functions make this even faster?

**Review by brandonroberts — APPROVED** [review]
🃏

**Pull request merged**